### PR TITLE
blend2d: migration to Conan v2 and fix compilation error in gcc12 with c++20

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **blend2d/***

- migratioon to Conan v2
- add sha256 in 0.0.18
- add test_v1_package
- fix compilation error in gcc12 with c++20 (backport from upstream)

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — APPROVED** [review]

File: recipes/blend2d/all/conandata.yml [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
     url: "https://blend2d.com/download/blend2d-beta17.zip"
     sha256: "06ee8fb0bea281d09291e498900093139426501a1a7f09dba0ec801dd340635e"
 
 patches:
   "0.0.18":
-    - base_path: "source_subfolder"
-      patch_file: "patches/0.0.17-0001-disable-embed-asmjit.patch"
+    - patch_file: "patches/0.0.17-0001-disable-embed-asmjit.patch"
+    # fix gcc12 compilation error, backport from upstream
+    - patch_file: "patches/0.0.18-0002-fix-gcc12-compilationerror.patch"
```
Comment (prince-chrismc) [inline recipes/blend2d/all/conandata.yml:13]: Could you please add the metadata to so we can track these better 🙏 

https://github.com/conan-io/conan-center-index/blob/master/docs/conandata_yml_format.md#patches-fields

File: recipes/blend2d/all/conanfile.py [hunk 2]
```diff
@@ -19,22 +27,11 @@ class Blend2dConan(ConanFile):
...
-
-    @property
-    def _is_msvc(self):
-        return str(self.settings.compiler) in ["Visual Studio", "msvc"]
 
     def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        tools.files.copy(self, "CMakeLists.txt", self.recipe_folder, self.export_sources_folder)
```
Comment (SpaceIm) [inline recipes/blend2d/all/conanfile.py:32]: this file is useless now, it should be removed

File: recipes/blend2d/all/CMakeLists.txt [hunk 3]
```diff
@@ -4,7 +4,4 @@ project(cmake_wrapper LANGUAGES CXX)
 set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
 set(CMAKE_CXX_STANDARD 11)
 
-include(conanbuildinfo.cmake)
-conan_basic_setup(KEEP_RPATHS)
-
-add_subdirectory(source_subfolder)
+add_subdirectory(src)
```
Comment (SpaceIm) [inline recipes/blend2d/all/CMakeLists.txt:7]: please remove this file and call CMake configuration on upstream CMakeLists directly.

You can inject CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS & CMAKE_CXX_STANDARD in the toolchain.

**Review by jgsogo — APPROVED** [review]
Let's merge this one, adding the `sha256` is important. Then we can come back to the recipe for other fixes.

**Pull request merged**